.github/workflows/dependency-security.yml

```
@@ -0,0 +1,44 @@
+    branches: [main]
+jobs:
+  audit-and-scan:
+    name: Dependency Audit & License Verification
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Codebase
+        uses: actions/checkout@v4
+      - name: Setup Node.js Environment
+        uses: actions/setup-node@v4
+        with:
+          node-with-registry: 'https://npmjs.org'
```

actionlint (1.7.12)

[error] input "node-with-registry" is not defined in action "actions/setup-node@v4". available inputs are "always-auth", "architecture", "cache", "cache-dependency-path", "check-latest", "mirror", "mirror-token", "node-version", "node-version-file", "registry-url", "scope", "token"

(action)